src/Lecture3.hs

```diff
@@ -134,10 +162,23 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
-
+  List1 x xs <> List1 y ys = List1 x $ xs ++ y:ys
+  -- just learned about $. not sure if this is a good use of it.
+  -- feels weird to think about applying the left side to the right side in this case
+  -- bc it's part of a constructor
 
 {- | Does 'List1' have the 'Monoid' instance? If no then why?
 
+hmm, we would need to identify a neutral element of polymorphic type a for the given
+mappend function of list concat. it would need to be appendable to either side of
+an arbitrary value of type List1 and not change the value. given that List1 needs
+at least 1 element in it, then it seems contradictory to append it to another List1 value
+and expect the value not to change -> the result would be a list of longer length.
+
+maybe we could introduce some sentinel value that we agree to ignore if we see it?
+we would need to do this for every type a though. maybe if we change List1 to be of
+type Maybe a instead...
```

Comment:
That's nice reasoning! 🔝 

In other simple words, there's no such thing as "empty non-empty list" 🙂